### PR TITLE
rehearse: do not decode templates

### PR DESCRIPTION
**Pull request by bbguimaraes**
Decoding the templates is not necessary for any operation in `rehearse`. Treating the files as an array of bytes is enough to tell if it changed and to populate the temporary `ConfigMap`s.

Not decoding simplifies the code and greatly simplifies the tests.

### CODE REVIEW

**Comment by bbguimaraes** [comment]
`OWNERS` in `vendor/`, didn't see that coming.

/uncc danwinship bparees
/cc @openshift/developer-productivity-test-platform

File: pkg/config/template.go [hunk 1]
```diff
@@ -40,24 +34,16 @@ const (
 )
 
 func getTemplates(templatePath string) (CiTemplates, error) {
-	templates := make(map[string]*templateapi.Template)
+	templates := make(CiTemplates)
 	err := filepath.Walk(templatePath, func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			return fmt.Errorf("prevent panic by handling failure accessing a path %q: %v", path, err)
+		if err != nil || info.IsDir() || !(strings.HasSuffix(path, ".yaml") || strings.HasSuffix(path, ".yml")) {
```
Comment (droslean) [inline pkg/config/template.go:39]: this check can be done with `isYAML()` as well.
```go
if err != nil || isYAML(path, info) {
```

**Review by droslean — COMMENTED** [review]
nice one.

File: pkg/config/template.go (hunk 1 above)
Comment (bbguimaraes) [reply pkg/config/template.go:39]: I inlined that function because it is only ever used here, but that got lost when I moved commits around. Updated.

Comment (droslean) [reply pkg/config/template.go]: why we want `(strings.HasSuffix(path, ".yaml")` to be false?

**Pull request merged**